### PR TITLE
Initial working code.

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+var gulp = require('gulp');
+var ts = require('gulp-typescript');
+var ts2dart = require('./');
```
Comment (mprobst) [inline gulpfile.js]: require ./? Maybe require('./index.js')?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        file.contents = new Buffer(ts2dart.translateFiles([file.path]));
+        file.path = file.path.replace(/.ts$/, '.dart');
+        done(null, file);
+    });
+}
+exports.transpile = transpile;
+function format() {
+    return spawn({ cmd: which.sync('dartfmt') });
+}
+exports.format = format;
```
Comment (mprobst) [inline index.js]: might be simpler to just assign exports.format = function() ... directly.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+var spawn = require('gulp-spawn');
+var through = require('through2');
+var ts2dart = require('ts2dart');
+var which = require('which');
+
+function transpile() {
+    return through.obj(function (file, enc, done) {
+        file.contents = new Buffer(ts2dart.translateFiles([file.path]));
```
Comment (mprobst) [inline index.js]: indentation.

File: gulpfile.js (hunk 1 above)
Comment (alexeagle) [inline gulpfile.js]: Actually I think this is right. We want other people to be able to just require('gulp-ts2dart') so I should be sure that including this directory works, not a particular file.

File: index.js (hunk 3 above)
Comment (alexeagle) [inline index.js]: Will hook up the formatter

File: index.js (hunk 2 above)
Comment (alexeagle) [inline index.js]: Yes, interestingly I had started out writing this in TypeScript but decided it was too much work to wire in the compiler, so this file is the tsc output of my original TS version.

**Comment by mprobst** [comment]
Otherwise LGTM.

**Pull request merged**